### PR TITLE
#13 Update to docker hub api v2

### LINKED ISSUE #13: Dockerhub API change breaks tag lookup
Looks like dockerhub changed their API / obsoleted version 1:
```
String dockerVersion = "exasol/docker-db:"
                + assertDoesNotThrow(
                () -> new ExasolVersionNumberProviderFactory()
                        .buildExasolVersionNumberProvider()
                        .getLatestReleaseForMinor(7, 1)
        );

...
java.io.FileNotFoundException: https://registry.hub.docker.com/v1/repositories/exasol/docker-db/tags
```

https://github.com/exasol/exasol-version-number-provider/blob/a178d86f25ab894a661a033c754a29d45c46b014/src/main/java/com/exasol/versionnumberprovider/ExasolVersionNumberProviderFactory.java#L20

Same URL with `/v2/` returns a result.

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #13

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -78,7 +89,7 @@
     <parent>
         <artifactId>exasol-version-number-provider-generated-parent</artifactId>
         <groupId>com.exasol</groupId>
-        <version>0.2.3</version>
+        <version>0.2.4</version>
         <relativePath>pk_generated_parent.pom</relativePath>
     </parent>
-</project>
+</project>
```
Comment (exaSR) [inline pom.xml]: Missing end of line :)
I recommend changing IDE settings to auto-add EOL at EOF

File: src/main/java/com/exasol/versionnumberprovider/dockerhub/Tag.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    /**
+     * Set the name of this tag.
+     * 
+     * @param name name of this tag
+     */
+    public void setName(final String name) {
+        this.name = name;
+    }
+}
```
Comment (exaSR) [inline src/main/java/com/exasol/versionnumberprovider/dockerhub/Tag.java]: EOF

File: src/test/java/com/exasol/versionnumberprovider/ExasolVersionNumberProviderFactoryTest.java [hunk 3]
```diff
@@ -1,18 +1,21 @@
...
     @Test
     void test() throws IOException {
         final ExasolVersionNumberProvider versionNumberProvider = new ExasolVersionNumberProviderFactory()
                 .buildExasolVersionNumberProvider();
-        assertThat(versionNumberProvider.getAllReleaseNumbers(), hasItems("7.0.0", "7.0.1", "7.0.2"));
+        final List<String> allReleaseNumbers = versionNumberProvider.getAllReleaseNumbers();
+        assertAll(() -> assertThat(allReleaseNumbers, hasItems("6.2.0-d1", "7.0.0", "7.0.1", "7.0.2", "7.1.14")),
+                () -> assertThat(allReleaseNumbers, hasSize(greaterThanOrEqualTo(57))));
     }
 }
```
Comment (exaSR) [inline src/test/java/com/exasol/versionnumberprovider/ExasolVersionNumberProviderFactoryTest.java]: EOF

File: src/test/java/com/exasol/versionnumberprovider/ExasolVersionNumberProviderTest.java [hunk 4]
```diff
@@ -4,28 +4,31 @@
...
         assertThat(versionNumberProvider.getLatestReleaseForMajor(6), equalTo("6.2.10-d1"));
     }
 
     @Test
     void testGetLatestForMinor() throws IOException {
-        final ExasolVersionNumberProvider versionNumberProvider = FACTORY.getLocalDockerHubResponseDump();
+        final ExasolVersionNumberProvider versionNumberProvider = FACTORY.getLocalDockerHubResponseDump(RESPONSE_FILE);
         assertThat(versionNumberProvider.getLatestReleaseForMinor(6, 1), equalTo("6.1.12-d1"));
     }
 }
```
Comment (exaSR) [inline src/test/java/com/exasol/versionnumberprovider/ExasolVersionNumberProviderTest.java]: EOF

File: src/test/resources/dockerhub_response.json [hunk 5]
```diff
@@ -1,198 +1,309 @@
...
+            "name": "7.1.10",
+            "repository": 1315215,
+            "full_size": 2271996619,
+            "v2": true,
+            "tag_status": "active",
+            "tag_last_pulled": "2022-09-27T13:05:11.809288Z",
+            "tag_last_pushed": "2022-05-06T06:30:26.288818Z"
+        }
+    ]
+}
```
Comment (exaSR) [inline src/test/resources/dockerhub_response.json]: EOF

File: src/test/resources/simple_dockerhub_response.json [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+            },
+            {
+                "name": "latest-6.2"
+            },
+            {
+                "name": "latest-7.0"
+            }
+        ]
+    ]
+}
```
Comment (exaSR) [inline src/test/resources/simple_dockerhub_response.json]: EOF

**Review by exaSR — DISMISSED** [review]
Thanks for the super-speedy fix, and sorry this turned out to be more complex than estimated at first glance.

Might want to fix those missing newlines at end of files...

**Review by ckunki — APPROVED** [review]

**Pull request merged**